packages/wrangler/e2e/startDevWorker.test.ts

```diff
@@ -19,6 +19,8 @@ let mf: Miniflare | undefined;
 let res: MiniflareResponse | undici.Response | undefined;
 let ws: undici.WebSocket | undefined;
 
+type OptionalKeys<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
```

Comment:
Not 100% this would work, but could help make the typing more concise yet explicit too. 
```suggestion
type Optional<T, K extends keyof T> = Partial<Record<K, T[K]>>;

type OptionalKeys<T, K extends keyof T> = Omit<T, K> & Optional<T, K>;
```